packages/react-dom/src/events/__tests__/SyntheticEvent-test.js

```diff
@@ -18,63 +17,122 @@ describe('SyntheticEvent', () => {
   var createEvent;
 
   beforeEach(() => {
-    // TODO: can we express this test with only public API?
-    SyntheticEvent = require('events/SyntheticEvent');
     React = require('react');
     ReactDOM = require('react-dom');
     ReactTestUtils = require('react-dom/test-utils');
-
-    createEvent = function(nativeEvent) {
-      var target = require('../getEventTarget')(nativeEvent);
-      return SyntheticEvent.getPooled({}, '', nativeEvent, target);
+    createEvent = (eventType, nativeEvent) => {
+      const defaultNativeEvent = {
+        bubbles: true,
+        cancelable: true,
+        ...nativeEvent,
+      };
+      var event = new Event([eventType, defaultNativeEvent]);
+      event.initEvent(
+        eventType,
+        defaultNativeEvent.bubbles,
+        defaultNativeEvent.cancelable,
+      );
+      return event;
     };
   });
 
   it('should normalize `target` from the nativeEvent', () => {
-    var target = document.createElement('div');
-    var syntheticEvent = createEvent({srcElement: target});
+    var click = jest.fn();
+    var container = document.createElement('div');
+
+    var onClick = e => click(e.target);
+
+    var instance = ReactDOM.render(<div onClick={onClick} />, container);
 
-    expect(syntheticEvent.target).toBe(target);
-    expect(syntheticEvent.type).toBe(undefined);
+    document.body.appendChild(container);
+
+    var event = createEvent('click', {srcElement: instance});
+    var elem = ReactDOM.findDOMNode(instance);
+    elem.dispatchEvent(event);
+    expect(click).toBeCalledWith(elem);
+    document.body.removeChild(container);
   });
 
   it('should be able to `preventDefault`', () => {
-    var nativeEvent = {};
-    var syntheticEvent = createEvent(nativeEvent);
+    var click = jest.fn();
+    var container = document.createElement('div');
 
-    expect(syntheticEvent.isDefaultPrevented()).toBe(false);
-    syntheticEvent.preventDefault();
-    expect(syntheticEvent.isDefaultPrevented()).toBe(true);
+    var onClick = e => {
+      click(e.isDefaultPrevented());
+      e.preventDefault();
+      click(e.isDefaultPrevented());
+    };
 
-    expect(syntheticEvent.defaultPrevented).toBe(true);
+    var instance = ReactDOM.render(<div onClick={onClick} />, container);
 
-    expect(nativeEvent.returnValue).toBe(false);
+    document.body.appendChild(container);
+
+    var event = createEvent('click', {srcElement: instance});
```

Comment:
Only the first test was checking for `srcElement`. There shouldn't be a need to add this to every test.